docs/source/user/supporting-elements.rst

```diff
@@ -1,20 +1,146 @@
+* SubstituteDistance, computed as the number of substitutions to convert one dataset into another dataset
+* AbsoluteDistance<T>, computed as the absolute difference between any two numeric scalars of type T
+* L1Distance<T>, computed as the L1 distance between any two numeric vectors of type T
+* L2Distance<T>, computed as the L2 distance between any two numeric vectors of type T
+
+A measure is a function for computing a number from a subset of a set.
+In OpenDP, a concrete example is the type `MaxDivergence<f64>`,
+read as "the max divergence metric where numbers are expressed in terms of 64-bit floats."
+The max divergence metric has distances that correspond exactly to epsilon in the pure definition of differential privacy.
+
+Some common measures are:
+
+* MaxDivergence<T>, which corresponds to pure-DP, where epsilon has type T
+* SmoothedMaxDivergence<T>, which corresponds to approximate-DP, where epsilon and delta have type T
+
+Consider that it is never necessary to actually evaluate these functions!
+The sole purpose of metrics and measures is to provide the units for distances.
 
 Relations
 ---------
+We assert the privacy properties of a Transformation or Measurement's function via a relation.
+The relation accepts an input distance `d_in` and an output distance `d_out`,
+where `d_in` and `d_out` are distances in terms of the input and output metric or measure.
+
+Let `d_X` be the input metric, `d_Y` be the output metric or measure,
+and `f` be the function in the Transformation or Measurement.
+
+The relation fails if there exists two members of the input domain, x and x', that satisfy two conditions:
+
+#. d_X(x, x') <= d_in. The inputs are d_in-close.
+#. d_Y(f(x), f(x')) > d_out. But the outputs aren't d_out-close!
+
+In plaintext:
+There should be no perturbation to the input (by at most distance `d_in`) that can significantly influence the output (greater than distance `d_out`).
+
+The typical perturbation is the adjustment made to a dataset by adding or removing a user.
+
+
+Distances
+---------
+
+It is critical that you supply the correct `d_in` for the relation,
+whereas you can use utilities in the library to search for the tightest `d_out`.
+
+If you are starting with a dataset the input metric of your measurement will likely be SymmetricDistance.
+In this case, `d_in` is the maximum number of records that an individual may contribute.
+
+For example, consider a dataset of student exam scores where we want to compute the DP-mean.
+If we know the class has three exams, then we know each student may contribute at most three records, so `d_in` should be 3.
+
+We can use this information in a measurement that casts, imputes, clamps, resizes, averages, and then adds noise.
+We can use the relation on this chained measurement to conduct a binary search to find `d_out`, where `d_in` = 3.
+
+It is also possible to use `binary_search_chain` to find a noise scale parameter that satisfies a given budget.
```

Comment:
Hoo boy, the reason for binary_search is gonna be really hard for people to understand! It's tough to explain, but I think we need some more verbiage about what's going on and why you need to do it.